### ISSUE TITLE
Add code documentation

### ISSUE THREAD

**Issue by SchrodingersGat** [inventree/InvenTree#4702]
### Body of the issue

Ref: https://github.com/inventree/InvenTree/issues/2493#issuecomment-1523121443

Ref: https://github.com/mkdocstrings/mkdocstrings

Build code documentation into the InvenTree docs automatically

**Comment by SchrodingersGat** [comment]
It might be an idea to add a decorator to functions we wish to specify as "API" functions (which we make an effort not to change), while some functions are marked as internal. Or mark internal functions with an underscore prefix, or similar.

Having an API of public functions would help to make them more discoverable for inclusion in generated reports, for example

**Comment by matmair** [comment]
@SchrodingersGat I think it makes sense to only expose things we want to be included specifically like plugin mixins.

**Comment by SchrodingersGat** [comment]
So you would suggest an "opt in" approach for which classes and methods we expose?

**Comment by SchrodingersGat** [comment]
not stale

**Comment by matmair** [comment]
Still important

**Comment by matmair** [comment]
still important

**Comment by wolflu05** [comment]
I think this is still relevant because this would reduce a lot of duplicate documentation in the docstrings and the documentation. 

The suggested library was now also already added as part of #4824 documentation.

**Comment by matmair** [comment]
I think we are doing this now as suggested, right  @SchrodingersGat @wolflu05 ?

**Comment by SchrodingersGat** [comment]
Yes, we are certainly doing more of it. I think that this can be closed out.
